benchexec/tablegenerator/util.py

```diff
@@ -139,6 +139,49 @@ def to_decimal(s):
         return None
 
 
+def print_decimal(d):
```

Comment:
**Suggestion (non-blocking)**: Don't use a util method `print_decimal`,
but implement an own type for decimals with the corresponding `__str__` method.

The current code seems to work really well, but you still have to remember to call `util.print_decimal(value)` instead of using the common `str(value)`. This is error-prone (and less beautiful).

You could avoid this by implementing a subtype of `Decimal` with the content of `print_decimal` as `__str__` method.

Reply:
Good idea, sounds interesting. It would certainly be safer. Not sure yet whether it is worth because we use all the arithmetic methods on `Decimal`s so we would need to add a lot of forwardings methods to our type. But I will keep this in mind for later maybe.

For this PR I checked all occurrences of `str()` in `columns.py`.

Reply:
Would this not be good enough?

```python
import decimal
class PointNotationDecimal(decimal.Decimal): 
   ...:     def __str__(self): 
   ...:         return util.print_decimal(self)  # better: move code here
   ...:
Decimal = PointNotationDecimal  # for a smaller changeset, if desired                           
```

> For this PR I checked all occurrences of str() in columns.py

I also went through it and didn't find any issues.

Reply:
Sorry, now I get it!
`PointNotationDecimal(1) + PointNotationDecimal(1)` would still produce a `Decimal`, of course.